hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionImplementation.java

```diff
@@ -348,11 +366,13 @@ public class ConnectionImplementation implements ClusterConnection, Closeable {
       }
       this.metaCache = new MetaCache(this.metrics);
 
-      this.rpcClient = RpcClientFactory.createClient(this.conf, this.clusterId, this.metrics);
+      this.rpcClient = RpcClientFactory.createClient(this.conf, this.clusterId, this.metrics,
+        connectionAttributes);
       this.rpcControllerFactory = RpcControllerFactory.instantiate(conf);
       this.rpcCallerFactory = RpcRetryingCallerFactory.instantiate(conf, connectionConfig,
         interceptor, this.stats, this.metrics);
-      this.asyncProcess = new AsyncProcess(this, conf, rpcCallerFactory, rpcControllerFactory);
+      this.asyncProcess = new AsyncProcess(this, conf, rpcCallerFactory, rpcControllerFactory,
+        Collections.emptyMap());
```

Comment:
we can remove this from AsyncProcess constructor now, right? we shouldn't be using any attributes on the AsyncProcess since it's global

Reply:
ya, that constructor has already been removed so this won't even compile